tokio-timer/src/atomic.rs

```diff
@@ -0,0 +1,81 @@
+pub use self::imp::AtomicU64;
+
+#[cfg(target_pointer_width = "64")]
+mod imp {
+    use std::sync::atomic::{AtomicUsize, Ordering};
+
+    #[derive(Debug)]
+    pub struct AtomicU64 {
+        inner: AtomicUsize,
+    }
+
+    impl AtomicU64 {
+        pub fn new(val: u64) -> AtomicU64 {
+            AtomicU64 {
+                inner: AtomicUsize::new(val as usize),
+            }
+        }
+
+        pub fn load(&self, ordering: Ordering) -> u64 {
+            self.inner.load(ordering) as u64
+        }
+
+        pub fn store(&self, val: u64, ordering: Ordering) {
+            self.inner.store(val as usize, ordering)
+        }
+
+        pub fn fetch_or(&self, val: u64, ordering: Ordering) -> u64 {
+            self.inner.fetch_or(val as usize, ordering) as u64
+        }
+
+        pub fn compare_and_swap(&self, old: u64, new: u64, ordering: Ordering) -> u64 {
+            self.inner.compare_and_swap(
+                old as usize, new as usize, ordering) as u64
+        }
+    }
+}
+
+#[cfg(not(target_pointer_width = "64"))]
```

Comment:
Add hint/todo: `Remove when https://doc.rust-lang.org/nightly/std/sync/atomic/constant.ATOMIC_U64_INIT.html is landed`

Reply:
I did this.